pyquil/api/_quantum_computer.py

```diff
@@ -70,6 +73,245 @@ def _get_flipped_protoquil_program(program: Program) -> Program:
+
+    # The next part is not rigorous the sense that we simply truncate to the desired
+    # number of qubits. The problem is that orthogonal arrays of a certain strength for an
+    # arbitrary number of qubits are not known to exist.
+    num_expts, num_qubits = flip_matrix.shape
+    if len(meas_qubits) != num_qubits:
+        flip_matrix = flip_matrix[0:int(num_expts), 0:int(len(meas_qubits))]
+
+    for flip_array in flip_matrix:
+        total_prog_symm = program.copy()
+        prog_symm = _flip_array_to_prog(flip_array, meas_qubits)
+        total_prog_symm += prog_symm
+        symm_programs.append(total_prog_symm)
+        flip_arrays.append(flip_array)
+
+    return symm_programs, flip_arrays
+
+
+def consolidate_symmetrization_outputs(outputs: List[np.ndarray],
+                                       flip_arrays: List[Tuple[bool]]) -> np.ndarray:
+    """
+    Given bitarray results from a series of symmetrization programs, appropriately flip output
+    bits and consolidate results into new bitarrays.
+
+    :param outputs: a list of the raw bitarrays resulting from running a list of symmetrized
+        programs; for example, the results returned from _measure_bitstrings
+    :param flip_arrays: a list of boolean arrays in one-to-one correspondence with the list of
+        outputs indicating which qubits where flipped before each bitarray was measured.
+    :return: an np.ndarray consisting of the consolidated bitarray outputs which can be treated as
+        the symmetrized outputs of the original programs passed into a symmetrization method. See
+        estimate_observables for example usage.
+    """
+    assert len(outputs) == len(flip_arrays)
+
+    output = []
+    for bitarray, flip_array in zip(outputs, flip_arrays):
+        if len(flip_array) == 0:
+            # happens when measuring identity.
+            # TODO: better way of handling identity measurement? (in _measure_bitstrings too)
+            output.append(bitarray)
+        else:
+            output.append(bitarray ^ flip_array)
+
+    return np.vstack(output)
+
+
+def _measure_bitstrings(qc, programs: List[Program], meas_qubits: List[int],
+                        num_shots: int = 600) -> List[np.ndarray]:
+    """
+    Wrapper for appending measure instructions onto each program, running the program,
+    and accumulating the resulting bitarrays.
+
+    :param qc: a quantum computer object on which to run each program
+    :param programs: a list of programs to run
+    :param meas_qubits: groups of qubits to measure for each program
+    :param num_shots: the number of shots to run for each program
+    :return: a len(programs) long list of num_shots by num_meas_qubits bit arrays of results for
+        each program.
+    """
+    results = []
+    for program in programs:
+        if len(meas_qubits) == 0:
+            # corresponds to measuring identity; no program needs to be run.
+            results.append(np.array([[]]))
+            continue
```

Comment:
We might want to remove this check since typically running a program without any measurements results in an empty bitstring along with a warning. 

In any case ```run_symmetrized_readout_new``` called with ```measure_qubits = 0``` won't trigger this condition since the input ```programs``` list will be empty.

Reply:
- [x] todo remove it